tests/unit/test_replay_manager.py

```diff
@@ -1,19 +1,81 @@
+    from tempfile import NamedTemporaryFile
+    import json
+
+    ringtest_baseconfig["inputs"] = {}
+    ringtest_baseconfig["inputs"]["spikeReplay"] = {
+        "module": "synapse_replay",
+        "input_type": "spikes",
+        "spike_file": input_spikes_file,
+        "delay": 0,
+        "duration": 50,
+        "node_set": "RingB",  # no limits!
+    }
+
+    with NamedTemporaryFile("w", suffix='.json', delete=False) as config_file:
+        json.dump(ringtest_baseconfig, config_file)
+    return config_file
 
 
 @pytest.mark.forked
-def test_replay_manager_sonata():
+def test_sonata_spikes_reader(input_spikes_file):
     from neurodamus.replay import SpikeManager, MissingSpikesPopulationError
-    spikes_sonata = SAMPLE_DATA_DIR / "out.h5"
 
-    timestamps, spike_gids = SpikeManager._read_spikes_sonata(spikes_sonata, "NodeA")
-    npt.assert_allclose(timestamps[:8], [0.1, 0.15, 0.175, 2.275, 3.025, 3.45, 4.35, 5.7])
-    npt.assert_equal(spike_gids[:8], [1, 3, 2, 1, 2, 3, 1, 2])
+    timestamps, spike_gids = SpikeManager._read_spikes_sonata(input_spikes_file, "RingA")
+    npt.assert_allclose(timestamps, [0.1, 0.15, 0.175, 2.275, 3.025, 3.45, 4.35, 5.7, 6.975, 7.725])
+    npt.assert_equal(spike_gids, [1, 3, 2, 1, 2, 3, 1, 2, 3, 1])
 
     # We do an internal assertion when the population doesnt exist. Verify it works as expected
     with pytest.raises(MissingSpikesPopulationError, match="Spikes population not found"):
-        SpikeManager._read_spikes_sonata(spikes_sonata, "wont-exist")
+        SpikeManager._read_spikes_sonata(input_spikes_file, "wont-exist")
+
+
+@pytest.mark.forked
+def test_synapse_replay_load(replay_sim_config_file, input_spikes_file):
+    from neurodamus import Neurodamus
+    from neurodamus.core.configuration import Feature
+    from neurodamus.core.configuration import SimConfig
+
+    Neurodamus(
+        replay_sim_config_file.name,
+        # restrict_node_populations=["RingB"],
+        restrict_features=[Feature.Replay],
+        disable_reports=True,
+        cleanup_atexit=False,
+    )
+
+    spikes_replay = SimConfig.stimuli["spikeReplay"]
+    assert spikes_replay['Target'] == 'RingB'
+    assert spikes_replay['Mode'] == 'Current'
+    assert spikes_replay['Pattern'] == 'SynapseReplay'
+    assert spikes_replay['Delay'] == 0.0
+    assert spikes_replay['Duration'] == 50.0
+    assert spikes_replay['SpikeFile'] == input_spikes_file
+
+
+@pytest.mark.forked
+def test_replay_stim_generated(replay_sim_config_file):
+    from neurodamus import Neurodamus
+    from neurodamus.core.configuration import Feature
+
+    n = Neurodamus(
+        replay_sim_config_file.name,
+        restrict_features=[Feature.Replay],
+        disable_reports=True,
+        cleanup_atexit=False,
+    )
+
+    edges_a_b = n.circuits.get_edge_manager("RingA", "RingB")
+    conn_1_1001 = next(edges_a_b.get_connections(1001, 1))
+    time_vec = conn_1_1001._replay.time_vec.as_numpy()
```

Comment:
here we are checking only the time at which the spikes occur. Should we check also the peak values? I am not sure this can be easily extracted

Reply:
For replay event `ReplayStim`, we only have the control of spike time.